ReportingServicesTools/Functions/CatalogItems/Set-RsSubscription.ps1

```diff
@@ -0,0 +1,148 @@
+            and pipe the results to Set-RsSubscription which will create a new subscription on each report that exists with the same name on the destination folder '/New Folder' located at Report Server 'http://remote-machine:8080/reportserver_sql16'
+
+
+        .EXAMPLE
+            $paths = Get-RsCatalogItems -ReportServerUri 'http://localhost/ReportServer_sql14' -RsFolder /Origin | Where-Object TypeName -eq "Report" | Select-Object -ExpandProperty Path
+            Get-RsSubscription -ReportServerUri 'http://localhost/ReportServer_sql14' -Path $paths | Set-RsSubscription -ReportServerUri 'http://remote-machine:8080/reportserver_sql16' -RsFolder '/New Folder'
+
+            Description
+            -----------
+            This command will establish a connection to the Report Server located at http://localhost/ReportServer_sql14 using current user's credentials get all the paths from all reports at '/Origin' folder.
+            Then it uses the $paths variable to get all existing subscriptions and pipe the results to Set-RsSubscription which will create a new subscription on each report that exists with the same name on the destination folder '/New Folder' located at Report Server 'http://remote-machine:8080/reportserver_sql16'
+    #>
+    
+    [cmdletbinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+    param
+    (
+        [string]
+        $ReportServerUri,
+        
+        [Alias('ReportServerCredentials')]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+        
+        $Proxy,
+
+        [Alias('ReportPath','ItemPath')]
+        [Parameter(ParameterSetName='Report', Mandatory=$True)]
+        [string]
+        $Path,
+
+        [Alias('Folder')]
+        [Parameter(ParameterSetName='Folder', Mandatory=$True)]
+        [string]
+        $RsFolder,
+
+        [Parameter(Mandatory = $True, ValueFromPipeline=$true)]
+        [object[]]
+        $Subscription
+    )
+    Begin
+    {
+        $Proxy = New-RsWebServiceProxyHelper -BoundParameters $PSBoundParameters
+    }
+    
+    Process
+    {
+        #region Input Validation
+        if ([System.String]::IsNullOrEmpty($Path) -and [System.String]::IsNullOrEmpty($RsFolder)) {
+            throw 'No Folder or report Path was specified! You need to specify -RsFolder or -Path variable.'
+        }
+        #endregion Input Validation
+
+        try {
+            $parameters = $null
+
+            foreach ($sub in $Subscription) {
+                if ($RsFolder) {
+                    $Path = "$RsFolder/$($sub.Report)"
+                }
+                
+                Write-Verbose "Validating if destination exists..."
+                
+                try {
+                    $object = $null
+                    if ([System.String]::IsNullOrEmpty($ReportServerUri)) {
```

Comment:
In order to avoid the condition here , you can just pass the $Proxy down , it will also support the connect function and any other like credentials an